### ISSUE TITLE
Flaky test: ReadEventsFromPartitionAsyncThrowsIfConsumerClosedBeforeRead

### ISSUE THREAD

**Issue by pakrym** [Azure/azure-sdk-for-net#11870]
```
The iterator should have indicated the consumer was closed.

Expected: instance of <Azure.Messaging.EventHubs.EventHubsException> and property Reason equal to ClientClosed

But was: <System.Threading.Tasks.TaskCanceledException: A task was canceled.

at Azure.Messaging.EventHubs.Core.CancellationTokenExtensions.ThrowIfCancellationRequested[T](CancellationToken instance) in /Users/runner/runners/2.166.4/work/1/s/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Core/CancellationTokenExtensions.cs:line 31

at Azure.Messaging.EventHubs.Consumer.EventHubConsumerClient.CloseAsync(CancellationToken cancellationToken) in /Users/runner/runners/2.166.4/work/1/s/sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/EventHubConsumerClient.cs:line 602

at Azure.Messaging.EventHubs.Tests.EventHubConsumerClientTests.<>c__DisplayClass26_0.<<ReadEventsFromPartitionAsyncThrowsIfConsumerClosedBeforeRead>b__1>d.MoveNext() in /Users/runner/runners/2.166.4/work/1/s/sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientTests.cs:line 483

--- End of stack trace from previous location where exception was thrown ---

at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter1.GetResult() at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func1 invoke)

at NUnit.Framework.Internal.ExceptionHelper.RecordException(Delegate parameterlessDelegate, String parameterName)>

```

cc @jsquire

**Comment by jsquire** [comment]
That's a new one.   Do you happen to have a test run that I can reference?

**Comment by pakrym** [comment]
https://dev.azure.com/azure-sdk/public/_build/results?buildId=383250&view=logs&jobId=9cfc3b0b-53b1-524f-85ed-0ef4880bef82

**Comment by jsquire** [comment]
Fixed by #11850
